gpii/node_modules/solutionsRegistry/src/js/filter-schema-settings.js

```diff
@@ -0,0 +1,35 @@
+/* eslint-env node */
+"use strict";
+var fluid = require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+fluid.registerNamespace("gpii.universal.solutionsRegistry");
+
+/**
+ *
+ * Filter a settings schema so that it only has checks ("properties") for the specific observed settings.  Used in
+ * the sanity checks of the solutions registry itself, and in settings validation middleware.
+ *
+ * @param {Object} observedSettings - The observed settings that we should use to filter the schema.
+ * @param {Object} settingsSchema - The schema to filter.
+ * @return {Object} The filtered schema.
+ *
+ */
+gpii.universal.solutionsRegistry.filterSchemaToSettings = function (observedSettings, settingsSchema) {
+    if (typeof observedSettings === "object") {
```

Comment:
What's the alternative to this - given the documented argument type is actually {Object} - this is also a fairly weak and unstable check that will let through stuff like null due to longstanding JS bugs. Best to be clearer about what the meaning is here

Reply:
I think I was in part guarding against missing "observed settings".  Now that I know about `fluid.jQueryStandalone.isEmptyObject`, I standardised to use that to ensure that the filtering works as expected for empty objects (and `undefined`, `null`, etc.).  I also updated the JSdocs to indicate that the param is optional.